corehq/apps/locations/views.py

```diff
@@ -407,12 +412,40 @@ def products_form(self):
 
         form = MultipleSelectionForm(
             initial={'selected_ids': self.products_at_location},
-            submit_label=_("Update Product List")
+            submit_label=_("Update Product List"),
+            prefix="products",
         )
         form.fields['selected_ids'].choices = self.all_products
         return form
 
     @property
+    @memoized
+    def all_users(self):
```

Comment:
That's correct.  I'm mostly just scared of `CommCareUser.by_domain(self.domain)`, but I suppose that if that's an issue, this UI will be unusable anyways.  I'm also using ES to back the other part of this - figuring out who is already assigned to this location, because I can't filter by `location_id` in couch, although I suppose that could also be done in-memory from the full list of users.  I could be talked in to switching to couch, but I think it makes sense to have the same data source backing the queries involved, so I'd probably switch those to in-memory filtering.  No strong preference on my end.

Aside: I should probably move this into the form with the other queries.